### ISSUE TITLE
Bangle2 / Settings  Hangs when scrolling up and down through long menus

### ISSUE THREAD

**Issue by hughbarney** [espruino/BangleApps#2382]
![download](https://user-images.githubusercontent.com/3630481/206899017-a087a758-a812-4339-a8af-3a045af6f0ae.png)
![7d20c736fe80cdcd385e932a052157caa4d9c435](https://user-images.githubusercontent.com/3630481/206899026-0b577787-4943-475d-bdfb-1779a726dbd6.png)
### Affected hardware version

Bangle 2

### Your firmware version

2.16.28

### The bug

Whilst scrolling up and down through the settings menus the Bangle2 hangs and becomes unresponsive.
When connected to the IDE there are no error messages.
The only way to recover is a long press reboot of the Bangle 2.

I been experiencing this kind of hang on all of the v2.16.x builds I have installed  (.0 .7 .25 and .28)

I have narrowed it down to a minimum configuration of 
  bootloader 0.55
  settings 0.55
  standard launcher 0.19
  lock widget 0.7
  anton clock 0.11
  simplest clock 0.6

I have reproduced multiple times with the standard launcher and the icon launcher as the launcher. Also with Anton Clock or Simplest clock as the default clock.  This tells me it is not a problem with the launchers or the clocks.  Using simplest clock eliminates any  issues with 'fast load'.

## To reproduce

Go into the settings menu
Select a long menu (eg Utils)
Scroll up and down the menus a few times.
Very quickly you will see the display become corrupted and part of a menu stuck in the widget area of the display.
At this point the Bangle might already be hung
Continue to scoll up and down (swiping) a few more times and between 1-6 times the Bangle2 will hang.
If connected to the IDE nothing is displayed and the console is unresponsive (full hang).
The only way out is to do a long button reset.

**Comment by hughbarney** [comment]
Attached id a backup file of config.
[issue_2382_banglejs_backup.zip](https://github.com/espruino/BangleApps/files/10201996/issue_2382_banglejs_backup.zip)

**Comment by thyttan** [comment]
>Attached id a backup file of config.
[issue_2382_banglejs_backup.zip](https://github.com/espruino/BangleApps/files/10201996/issue_2382_banglejs_backup.zip)

I restored the backup but I haven't managed to reproduce the bug.

**Comment by hughbarney** [comment]
I have reproduced quite easily on a 2nd Bangle 2 now.
Maybe I need to make a short video.

**Comment by thyttan** [comment]
>I also noted that installing swipe menus made the issue go away, then I could uninstall swipe menu and it would come back.

This really makes me think it's something with `E.showscroller` that isn't updated with the latest changes, as suggested on [the forum thread](https://forum.espruino.com/comments/16794331/).

**Comment by BartS23** [comment]
I was able to reproduce the bug.

I have narrowed it down to the following:
```
var menu = {
  '< Back': () => print
};

for (n = 0; n <= 10; n++) {
  menu[n] = {};
}

E.showMenu(menu);
```

If you enter the following lines in the left side of the IDE, the watch will hang/reboot
```
Bangle.emit("drag", { "x": 0, "y": 0, "b": 1, "dx": 0, "dy": -250 })
Bangle.emit("drag", { "x": 0, "y": 0, "b": 1, "dx": 0, "dy": 167 })
```

The bug occurs only when Back is set.

**Comment by gfwilliams** [comment]
Thanks for the report - and thanks @BartS23 for such a concise way to reproduce this - I'll have a look and see if I can find/fix this this morning.

Just to add I managed to reproduce with the standard launcher too (as you'd expect given showScroller). Given the hang it may be related to the scrolling code - it's possible it's related to the LCD speed improvements that went into 2v16

**Comment by hughbarney** [comment]
I have tested and the  hang bug is fixed in 2.16.33 but I still get screen corruption in the settings app on scrolling through long menus.  I will log a new bug for that.
